server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -380,14 +381,14 @@ public CreateIndexRequest source(BytesReference source, XContentType xContentTyp
     public CreateIndexRequest source(Map<String, ?> source) {
         for (Map.Entry<String, ?> entry : source.entrySet()) {
             String name = entry.getKey();
-            if (SETTINGS.match(name)) {
+            if (SETTINGS.match(name, LoggingDeprecationHandler.INSTANCE)) {
```

Comment:
Bleh. These are problematic because `CreateIndexRequest` really feels like it shouldn't need stuff "from the server". At some point we'd like to extract it for the high level rest client.

Can you make a `source(Map<String, ?> source, DeprecationHandler deprecationHandler)` version of this method and make this one delegate to it? That doesn't solve the problem but I think it makes it more obvious.

Reply:
And when we go to actually solve the problem is gives us a potential way forward.